pytorch_lightning/strategies/strategy.py

```diff
@@ -84,6 +85,27 @@ def precision_plugin(self) -> PrecisionPlugin:
     def precision_plugin(self, precision_plugin: Optional[PrecisionPlugin]) -> None:
         self._precision_plugin = precision_plugin
 
+    @property
+    def optimizers(self) -> List[Optimizer]:
+        return self._optimizers
+
+    @optimizers.setter
+    def optimizers(self, optimizers: List[Optimizer]) -> None:
+        self._optimizers = optimizers
+        self._lightning_optimizers.clear()
+
+    @property
+    def _lightning_optimizers(self) -> Dict[int, LightningOptimizer]:
+        if not self._cached_lightning_optimizers:
+            # we create this on-the-fly instead of only in the optimizers setter because deleting this reference
+            # after every batch is supported as indicated by the test:
+            # tests/core/test_lightning_optimizer.py::test_lightning_optimizer_keeps_hooks
```

Comment:
I don't think doing this is sensible.

The test in question was added in #https://github.com/PyTorchLightning/pytorch-lightning/pull/6305 but I don't think it intended that deleting the lightning_optimizers per batch is supported.

We can also update the test and move this back to `optimizers.setter`

Up for discussion